06.bowling_object/lib/game.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require_relative 'frame'
+
+class Game
+  attr_reader :frames
+
+  def initialize(shot_marks)
+    delimited_marks = parse_marks(shot_marks.split(','))
+    @frames = delimited_marks.map { |m| Frame.new(m) }
+  end
+
+  def score
+    pin_score = @frames.map(&:score).sum
+    strike_bonus = calculate_strike_bonus
+    spare_bonus = calculate_spare_bonus
+    pin_score + strike_bonus + spare_bonus
+  end
+
+  private
+
+  def parse_marks(all_marks)
+    frames = []
+    frame = []
+    all_marks.each do |mark|
+      return [*frames, all_marks[(all_marks.size - frames.flatten.size) * -1..].dup] if last_frame?(frames)
+
+      frame << mark
+      if frame_terminated?(frame)
+        frames << frame.dup
+        frame.clear
```

Comment:
```suggestion
        frames << frame
        frame = []
```

の方がシンプルで良いと思います。（無駄にインスタンスを使い回さない）

Reply:
parse_marks メソッドを廃止し、initialize と build_framesに分けて実装を ```current_frame に Frame.new をセットする``` ように修正しました。
指摘内容と1対1になっておりませんがご承知おきください。